service.go

```diff
@@ -315,6 +340,12 @@ func (b *Builder) WithSIGHUP(handler func()) *Builder {
 	return b
 }
 
+// WithFields sets key/value structured log fields to be passed to the logger.
+func (b *Builder) WithFields(fields map[string]interface{}) *Builder {
```

Comment:
one final suggestion to rename the function in order to be more specific.
```suggestion
func (b *Builder) WithLogFields(fields map[string]interface{}) *Builder {
```

Reply:
done